core/src/test/java/com/linecorp/armeria/internal/common/ContextFutureCallbackArgumentsProvider.java

```diff
@@ -27,225 +29,149 @@
 
 import com.google.common.util.concurrent.MoreExecutors;
 
+import com.linecorp.armeria.common.RequestContext;
+
 public class ContextFutureCallbackArgumentsProvider implements ArgumentsProvider {
+    static class CallbackResult {
+        AtomicReference<RequestContext> context = new AtomicReference<>();
+        AtomicBoolean called = new AtomicBoolean();
+    }
+
+    private Function<CallbackResult, Void> fn = result -> {
```

Comment:
I guess we can use `Consumer<CallbackResult>` instead of `Function<CallbackResult, Void>`.
```suggestion
    private final Consumer<CallbackResult> fn = result -> {
```

Reply:
All the `thenApply`, `thenApplyAsync` etc. needs a return value, using `Function` can save us to have something like 

```
future.thenApply(res -> {
    fn.accept(result);
    return null;
})
                    
```

with `Function` we can simply use `future.thenApply(res -> fn.apply(result))`

Reply:
Other comments are addressed, thanks for reviewing

Reply:
> with Function we can simply use future.thenApply(res -> fn.apply(result))

I didn't think of the detail. Let's keep it as it is.